.githooks/check

```diff
@@ -1,26 +1,38 @@
-#!/bin/sh
+#!/bin/env bash
```

Comment:
I feel like we're reinventing test runners :(  But I don't know a simpler approach. Shell not propagating errors nicely despite `set -e` is definitely frustrating

Reply:
tbh, I am not liking my refactoring.

I might revert the effort I put into making the githooks and actions share the same source of truth. I just don't think it is worth the effort :(